src/shared/Microsoft.Git.CredentialManager/GitConfiguration.cs

```diff
@@ -267,18 +267,23 @@ public IEnumerable<string> GetAll(string name)
                 switch (git.ExitCode)
                 {
                     case 0: // OK
+                        string[] entries = data.Split('\0');
+
+                        // Because each line terminates with the \0 character, splitting leaves us with one
+                        // bogus blank entry at the end of the array which we should ignore
```

Comment:
I was thinking that what you really wanted was:

```cs
string[] entries = data.Split('\0', StringSplitOptions.RemoveEmptyEntries);
```

but you actually _do_ care about some empty values in the middle. This is an appropriate workaround.

Reply:
Yep. Entry entries are important. Terminating with `\0` would be useful if writing my own parser over the data, but not when using the `string.Split` method, sadly.